cos-fleetshard-operator-camel-it/src/test/java/org/bf2/cos/fleetshard/operator/it/camel/CamelConnectorReifyWithTypeTraitTest.java

```diff
@@ -0,0 +1,48 @@
+package org.bf2.cos.fleetshard.operator.it.camel;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import io.quarkiverse.cucumber.CucumberOptions;
+import io.quarkiverse.cucumber.CucumberQuarkusTest;
+import io.quarkus.test.junit.QuarkusTestProfile;
+import io.quarkus.test.junit.TestProfile;
+
+import static org.bf2.cos.fleetshard.support.resources.Resources.uid;
+
+@CucumberOptions(
+    features = {
+        "classpath:CamelConnectorReifyWithTypeTrait.feature"
+    },
+    glue = {
+        "org.bf2.cos.fleetshard.it.cucumber",
+        "org.bf2.cos.fleetshard.operator.it.camel.glues"
+    })
+@TestProfile(CamelConnectorReifyWithTypeTraitTest.Profile.class)
+public class CamelConnectorReifyWithTypeTraitTest extends CucumberQuarkusTest {
+    public static class Profile implements QuarkusTestProfile {
+        @Override
+        public Map<String, String> getConfigOverrides() {
+            final String ns = "cos-camel-" + uid();
+
+            final Map<String, String> configs = new HashMap<>();
+
+            configs.put("cos.namespace", ns);
+
+            configs.put(
+                "cos.operator.camel.connectors.traits.\"trait.camel.apache.org/container.image\"",
+                "quay.io/foo/bar:latest");
+            configs.put(
```

Comment:
The "container.image" trait is being added both ways (as a trait and as a types.trait), so the test doesn't really know if the "types" one is working. Or was this intended and I misinterpreted?

Reply:
The trait configuration has two levels:

1. global traits  `cos.operator.camel.connector.traits[trait_name] = value`
2. connector type specific trait }cos.operator.camel.connector.traits.types[type_id][trait_name] = value`

The test here is meant to verify that a connector type specific trait has a priority vs the global trait. 
This way we can configure i.e. the global memory limit for all the connectors except for some specific connector types where we can define an ad-hoc value

Reply:
Ah didn't see the bar vs baz. Ok :)